src/test/scala/io/iohk/ethereum/jsonrpc/server/http/JsonRpcHttpServerSpec.scala

```diff
@@ -118,9 +237,45 @@ class JsonRpcHttpServerSpec extends AnyFlatSpec with Matchers with ScalatestRout
       override val jsonRpcController = mockJsonRpcController
       override val jsonRpcHealthChecker = mockJsonRpcHealthChecker
 
+      override val ipTrackingEnabled: Boolean = config.ipTrackingEnabled
+      override val minRequestInterval: FiniteDuration = config.minRequestInterval
+      override val latestTimestampCacheSize: Int = config.latestTimestampCacheSize
+      override val latestRequestTimestamps = new LruMap[RemoteAddress, Long](latestTimestampCacheSize)
+
       def run(): Unit = ()
 
       override def corsAllowedOrigins: HttpOriginMatcher = HttpOriginMatcher(corsAllowedOrigin)
     }
+
+    val mockJsonRpcHttpServerWithIpRestriction = new JsonRpcHttpServer {
+      override val jsonRpcController = mockJsonRpcController
+      override val jsonRpcHealthChecker = mockJsonRpcHealthChecker
+
+      override val clock: Clock = FakeClock
+
+      override val ipTrackingEnabled: Boolean = true
+      override val minRequestInterval: FiniteDuration = config.minRequestInterval
+      override val latestTimestampCacheSize: Int = config.latestTimestampCacheSize
+      override val latestRequestTimestamps = new LruMap[RemoteAddress, Long](latestTimestampCacheSize)
+
+      def run(): Unit = ()
+
+      override def corsAllowedOrigins: HttpOriginMatcher = config.corsAllowedOrigins
+    }
+  }
+}
+
+object FakeClock extends Clock {
```

Comment:
Maybe this should be a `class FakeClock extends Clock` and on the TestSetup we create an instance of it instead of using this singleton?